### ISSUE TITLE
sql: investigate failures of big test

### ISSUE THREAD

**Issue by jordanlewis** [cockroachdb/cockroach#47206]

**Comment by yuzefovich** [comment]
I believe we started seeing these "already using disk" errors on big test only because we started running more tests as a part of "big test" (see #46722) and because we have a low limit for "disk" storage (100MB, and we're using in-memory storage engine for temp storage).

**Comment by yuzefovich** [comment]
Last night's [run](caused by non-values node passed as fixed value to zigzag join) crashed with `caused by non-values node passed as fixed value to zigzag join`
```
 5789 [19:51:25] :› ›  [run] {"Time":"2020-04-16T15:51:24.825872198-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/sql/logictest","Test":"TestSqlLiteLogic/l      ocal/slt_good_1.test#09","Output":"\tpanic: panic while executing 1 statements: SELECT _ FROM _ WHERE ((((_ = _) OR ((_ = _) AND (((((_ \u003e= _) AND (_ \u003c _)) AND ((_ IN       (_, _)) OR ((_ \u003c= _) AND (((_ \u003e _) AND (((((_ IN (_, _, __more2__)) AND ((((_ \u003e _) AND (_ \u003e _)) AND (_ \u003e= _)) OR (((_ \u003e= _)) AND (_ \u003e= _))))       OR (((_ BETWEEN _ AND _)))) OR (_ = _)))) OR ((((_ \u003e _) OR (((((_ \u003e _) AND (_ \u003e= _))) AND ((_ \u003c _) AND (_ \u003e _))) AND (_ \u003c _))))))))) OR ((_ \u003e       _) AND (_ \u003c= _)))))) OR (((_ \u003e _) AND (_ IS _)) AND (_ \u003c= _))) OR ((_ IS _) AND (_ \u003e _))) OR ((_ = _) AND (_ IS _)); caused by non-values node passed as fixed value to zigzag join\n"}^M
```

**Comment by yuzefovich** [comment]
So did the run from 3 days ago.

4, 5, and 6 days ago though it crashed because of OOM:
```
 5899 [05:56:11] :› ›  [run] {"Time":"2020-04-14T01:48:03.157814166-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/sql/logictest","Test":"TestSqlLiteLogic/l      ocal/slt_good_1.test#09","Output":"fatal error: runtime: out of memory\n"}^M
```

Ok, so my PR that attempts to resolve this issue is not to blame for the OOMs I observed.

**Comment by asubiotto** [comment]
Maybe we should rename this issue or create a new issue to track these failures. Why aren't these failures getting generated automatically?

**Comment by yuzefovich** [comment]
Renamed. Also see #47566 - something is wrong when trying to post an issue about big test failure.

**Comment by yuzefovich** [comment]
Current status of this issue is that I observed two problems when running big test on gceworker:
- when running with `-show-sql`, it got fatal OOM, and I filed #47976. I believe that issue to be the cause of OOM crashes we have seen
- when running with parallelization, I got `panic: non-values node passed as fixed value to zigzag join: *sql.unaryNode [recovered]`. Without disabling parallelization, it's hard to say which test triggers it. So I'll postpone looking into this problem. I guess it is likely to be an issue with planning, but let's wait for #47976 to be resolved. cc @RaduBerinde
